posthog/api/test/notebooks/test_notebook.py

```diff
@@ -0,0 +1,267 @@
+        ]
+
+        response = self.client.patch(
+            f"/api/projects/{self.team.id}/notebooks/{short_id}",
+            {"short_id": "something else"},
+        )
+        assert response.status_code == status.HTTP_200_OK
+        assert response.json()["short_id"] == short_id
+
+    def test_filters_based_on_params(self):
+        other_user = User.objects.create_and_join(self.organization, "other@posthog.com", "password")
+        needle_notebook = Notebook.objects.create(team=self.team, title="needle", created_by=self.user)
+        delivery_van_notebook = Notebook.objects.create(team=self.team, title="Delivery van", created_by=self.user)
+        other_users_notebook = Notebook.objects.create(team=self.team, title="need to know", created_by=other_user)
+
+        results = self.client.get(
+            f"/api/projects/{self.team.id}/notebooks?search=needl",
+        ).json()["results"]
+
+        assert len(results) == 1
+        assert results[0]["short_id"] == needle_notebook.short_id
+
+        results = self.client.get(
+            f"/api/projects/{self.team.id}/notebooks?search=need",
+        ).json()["results"]
+
+        assert len(results) == 2
+        assert results[0]["short_id"] == other_users_notebook.short_id
+        assert results[1]["short_id"] == needle_notebook.short_id
+
+        results = self.client.get(
+            f"/api/projects/{self.team.id}/notebooks?user=true",
+        ).json()["results"]
+
+        assert len(results) == 2
+        assert results[0]["short_id"] == delivery_van_notebook.short_id
+        assert results[1]["short_id"] == needle_notebook.short_id
+
+        results = self.client.get(
+            f"/api/projects/{self.team.id}/notebooks?created_by={other_user.id}",
+        ).json()["results"]
+
+        assert len(results) == 1
+        assert results[0]["short_id"] == other_users_notebook.short_id
+
+    def test_listing_does_not_leak_between_teams(self):
+        another_team = Team.objects.create(organization=self.organization)
+        another_user = User.objects.create_and_join(another_team.organization, "other@example.com", password="")
+
+        self.client.force_login(another_user)
+        response = self.client.post(
+            f"/api/projects/{another_team.id}/notebooks", data={"title": f"other_team_notebook"}
+        )
+        assert response.status_code == status.HTTP_201_CREATED
+
+        self.client.force_login(self.user)
+        response = self.client.post(f"/api/projects/{self.team.id}/notebooks", data={"title": "this_team_notebook"})
+        assert response.status_code == status.HTTP_201_CREATED
+
+        response = self.client.get(f"/api/projects/{self.team.id}/notebooks")
+        assert response.status_code == status.HTTP_200_OK
+        assert response.json()["count"] == 1
+        assert response.json()["results"][0]["title"] == "this_team_notebook"
+
+    @skip("is this not how things work?")
+    def test_creating_does_not_leak_between_teams(self):
```

Comment:
@Twixes am I being silly

I can create insights & notebooks in a different team. Do we need a manual protection, or is the test setup wrong, or my assumptions wrong?

Reply:
I think there should be some built-in protection, but can't point where without a bit of digging – but I see you sorted this out in some way? 😄